### PR TITLE
Add a more "business useful" example function

### PR DESCRIPTION
**Pull request by jskeet**
This is intended to show something more than "hello world", partly
to get the feeling of what real function development is like.

### CODE REVIEW

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+perform time zone conversions based on the [IANA time zone
+database](https://www.iana.org/time-zones).
+
+## Parameters
+
+All input is provided via the following query parameters:
+
+- `fromZone`: The ID of the time zone to convert from
+- `toZone`: The ID of the time zone to convert to
+- `when`: The local date/time to convert, from `fromZone` to `toZone`
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md:13]: Local here means fromZone, right? I would say so explicitly, here and everywhere.

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    The result is provided by shifting the input value by the length of the "gap" in local time (usually one hour).
+
+## Examples
+
+To try this function out, run it locally on port 8080 and use the following URLs:
+
+- http://localhost:8080/?when=2020-04-09T08:50:00.123&fromZone=Europe/London&toZone=Europe/Paris
+  This converts an unambiguous input from London to Paris.
+- http://localhost:8080/?when=2020-03-29T01:50:00.123&fromZone=Europe/London&toZone=Europe/Paris
+  This convers an input which is skipped due to "spring forward in March", converting it to Paris.
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md:56]: nit: conver**t**s
nit: converts **a London** input...

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- http://localhost:8080/?when=2020-10-25T01:50:00.123&fromZone=Europe/London&toZone=Europe/Paris
+  This converts an ambiguous input due to "fall back in October" in London, but converting to Paris which "falls back" at the same time- so the result is unambiguous.
+- http://localhost:8080/?when=2020-10-25T01:50:00.123&fromZone=Europe/London&toZone=Europe/Paris
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md:59]: toZone is still Paris, should be Los Angeles according to the comment below.

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+        /// <summary>
+        /// Constructs a converter using the given time zone provider and source.
+        /// </summary>
+        /// <param name="provider">The time zone provider to use.</param>
+        public TimeZoneConverter(IDateTimeZoneProvider provider) =>
+            _timeZoneProvider = provider;
+
+        /// <summary>
+        /// Logic for your function goes here.
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs:56]: nit: Maybe change this now.

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                return CreateResult(converted, ConversionType.Unambiguous);
+            }
+
+            ConversionResult ConvertAmbiguousInput()
+            {
+                var earlyConverted = mapping.First().WithZone(toZone).LocalDateTime;
+                var lateConverted = mapping.Last().WithZone(toZone).LocalDateTime;
+                return CreateResult(
+                    // While it would be very odd for the results of converting the earlier instant to be
+                    // later than the results of the converting the later instant, it's theoretically possible.
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs:145]: nit: than the results of _(the)_ converting

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md (hunk 1 above)
Comment (jskeet) [reply examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md:13]: It's slightly tricky. The value itself isn't in any zone; it's *interpreted* with respect to `fromZone`. Will have a think about another way of expressing that.

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md (hunk 2 above)
Comment (jskeet) [reply examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md:56]: Thanks - or something similar.

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md (hunk 3 above)
Comment (jskeet) [reply examples/Google.Cloud.Functions.Examples.TimeZoneConverter/README.md:59]: Doh - that's odd. Will fix, then check all of them again.

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs (hunk 4 above)
Comment (jskeet) [reply examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs:56]: Ha! Yes indeed :)

File: examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs (hunk 5 above)
Comment (jskeet) [reply examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs:145]: Done, thank.s

**Pull request merged**